### ISSUE TITLE
Decimal trimmed when CSV import

### ISSUE THREAD

**Issue by Bennoo** [Azure/azure-documentdb-datamigrationtool#12]
Hi there, 
Using the tool I encountered an issue during import of a CSV file.
This is the sample from the _original CSV_:
`0.485595703125,0.8212890625,0.0355102539062,62`

This is imported in the _DocumentDB_ as:
`485595703125,8212890625,3551025390625,62`

As you can see, the 0's are purely trimmed. Do you have an idea of the issue?

Thanks,

**Comment by nahk-ivanov** [comment]
Can you please tell us what your system locale is? Current CSV parser implementation is using the system-wide decimal separator. Maybe you have comma there, instead of a dot and that causes it to fail. I tried your sample values on my en-us machine and everything was good. Here's the data I used:

```
"Column1","Column2","Column3","Column4"
0.485595703125,0.8212890625,0.0355102539062,62
```

and result I got was

```
{
  "Column1": 0.485595703125,
  "Column2": 0.8212890625,
  "Column3": 0.0355102539062,
  "Column4": 62,
  "id": "bb6e7cc9-afba-460f-a9f4-c5484090c1d3"
}
```

There is a long story behind this, we can discuss the solution that will work for you, once confirmed that it is the root cause.

**Comment by Bennoo** [comment]
Thanks for your answer!
Yes my system is in Fr-be. And indeed setting my machine in En-US is solving the issue!!
Is there a possibility to provide a supplementary parameter to the console command in order to ask for an another region setting?

Many thanks,

**Comment by nahk-ivanov** [comment]
When implementing this feature we tried our best to support multiple scenarios. The original problem was that Excel export uses system-wide setting for CSVs. This is why we went with a "hybrid" approach, where we always use `comma` as a values separator (it is a **C**SV, after all), but system locale for the values (to be a little bit more aligned with other Microsoft products). I see that it can potentially fail when you have `comma` as a decimal separator, but feed it a file with a dot-separated decimals. In this case your system will most likely have a different _list separator_ as well and, unfortunately, Excel will use system-wide list separator when exporting into CSV, so it can look like `1|0,2|3`, which in my opinion is not a CSV anymore.

We could go with system locale completely (like Excel does), but that would not help in your case either, without switching the locale. Thus, to make our parser a little bit more "agile" we decided to implement this "hybrid" approach, where values should be separated by `commas`, but actual values are parsed with the system-wide regional settings in mind. In this case we are fully covered for standard `en-us` and _maybe_ will not fail for some other locales.

Now coming to the solution, maybe we should introduce a new setting **Use system regional settings** for CSV imports, which, when checked, will use _list separator_ as well, and when unchecked - use invariant culture for values and hard-coded `comma` as a values separator.

I will leave the issue open until we add the setting. Meanwhile, the only workaround I can suggest is to switch the locale to `en-us`, import your data and then switch back.
